src/pages/distroSettings/tabs/GeneralTab/DeleteDistro/index.tsx

```diff
@@ -0,0 +1,121 @@
+import { useState } from "react";
+import { useMutation, useQuery } from "@apollo/client";
+import Button from "@leafygreen-ui/button";
+import Tooltip from "@leafygreen-ui/tooltip";
+import { Description } from "@leafygreen-ui/typography";
+import { Field } from "@rjsf/core";
+import { useParams } from "react-router-dom";
+import { ConfirmationModal } from "components/ConfirmationModal";
+import ElementWrapper from "components/SpruceForm/ElementWrapper";
+import { useToastContext } from "context/toast";
+import {
+  DeleteDistroMutation,
+  DeleteDistroMutationVariables,
+  UserDistroSettingsPermissionsQuery,
+  UserDistroSettingsPermissionsQueryVariables,
+} from "gql/generated/types";
+import { DELETE_DISTRO } from "gql/mutations";
+import { USER_DISTRO_SETTINGS_PERMISSIONS } from "gql/queries";
+
+interface ModalProps {
+  closeModal: () => void;
+  open: boolean;
+  distroId: string;
+}
+
+const Modal: React.FC<ModalProps> = ({ closeModal, distroId, open }) => {
+  const dispatchToast = useToastContext();
+
+  const [deleteDistro] = useMutation<
+    DeleteDistroMutation,
+    DeleteDistroMutationVariables
+  >(DELETE_DISTRO, {
+    onCompleted() {
+      dispatchToast.success(
+        `The distro “${distroId}” was deleted. Future visits to this page will result in an error.`
+      );
+    },
+    onError(err) {
+      dispatchToast.error(
+        `There was an error deleting the distro: ${err.message}`
```

Comment:
I've felt lately that our error toasts can get pretty verbose, thoughts about including just `err.message` as the content? (The toast title is already "Error!").